hermes/hermes.go

```diff
@@ -0,0 +1,95 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// hermes.go defines the structures necessary for the main hermes object to
+// monitor a storage system.
+
+package hermes
+
+import (
+	"github.com/google/cloudprober"
+)
+
+// FileOperation is an int used as part of the FileOperation enum within the intent log of Hermes' StateJournal.
+type FileOperation int
+
+// FileOperation enum is used for marking the operation intent within the StateJournal of Hermes.
+// FileOperation has two possible values: CREATE, DELETE.
+const (
+	Create FileOperation = iota
+	Delete
+)
+
+// String() allows the FileOperation constants to be conveniently converted to a print-friendly format.
+// Returns:
+// - string: Returns the print-friendly string version of the fileOperation enum.
+func (fileOperation FileOperation) String() string {
+	return [...]string{"Create", "Delete"}[fileOperation]
+}
+
+// Intent stores the next intended file operation of Hermes.
+// This is used as part of the StateJournal of Hermes.
+type Intent struct {
+	operation FileOperation `json:"fileoperation"` // Stores the file operation intent, either CREATE or DELETE
+	filename  string        `json:"filename"`      // Stores the filename that the operation is being performed on.
+}
+
+// StateJournal stores the state of Hermes in two parts: the next operation intent and a map of filenames.
+// The intent stores the operation to be performed and the name of the file that the operation is being performed on.
+// The filenames map is a map of file IDs to filenames.
+// If an entry does not exist for a given ID, then the file does not exist (i.e. has been deleted)
+type StateJournal struct {
+	// intent stores the next intended file operation and the name of the file that the operation is being performed on.
+	intent Intent `json:"intent"`
+	// filenames is a map of file IDs to filenames.
+	filenames map[int]string `json:"filenames"`
+}
+
+// Init initialises the map in the StateJournal so that entries can be added to it.
+func (sj *StateJournal) Init() {
+	sj.filenames = make(map[int]string) // initialise filenames to a map
+}
+
+// Hermes is the main Hermes prober that will startup Hermes and initiate monitoring targets.
+type Hermes struct {
+	// Journal stores the state of Hermes as a combination of next operation intent and a filenames map
+	Journal StateJournal
+	// CancelCloudprober is a cancel() function associated with the context passed to Cloudprober when initialised.
+	CancelCloudprober func()
+}
+
+// InitialiseCloudproberFromConfig initialises Cloudprober from the config passed as an argument.
+// Parameters:
+// - config: config should be the contents of a Cloudprober config file. This is most likely: "grpc_port=9314"
+//           -> the "grpc_port:" field is the only required field for the config.
```

Comment:
The example above uses "=" as the separator and here we have ":". Which ones is correct?

Reply:
":" is correct, but that has been removed as it is no longer necessary in this file.